stdlib/src/builtin/string_literal.mojo

```diff
@@ -530,28 +530,28 @@ struct StringLiteral(
 
     @always_inline
     fn as_bytes(self) -> Span[Byte, StaticConstantOrigin]:
-        """
-        Returns a contiguous Span of the bytes owned by this string.
+        """Returns a contiguous slice of bytes.
 
         Returns:
-            A contiguous slice pointing to the bytes owned by this string.
-        """
+            A contiguous slice pointing to bytes.
 
+        Notes:
+            This does not include the trailing null terminator.
+        """
         return Span[Byte, StaticConstantOrigin](
             ptr=self.unsafe_ptr(), length=self.byte_length()
         )
 
     @always_inline
     fn as_bytes(ref self) -> Span[Byte, __origin_of(self)]:
-        """Returns a contiguous slice of the bytes owned by this string.
+        """Returns a contiguous slice of bytes.
```

Comment:
I'm not sure I see how this is better. The bytes are owned by the literal.
> Returns a contiguous slice of bytes.
doesn't really describe what the bytes contain at all.

Reply:
It's to keep the documentation consistent across owning and non-owning types. 

> doesn't really describe what the bytes contain at all.

"utf-8 bytes" sounds less confusing 4u?

Reply:
My point was that "owned by this string" was descriptive about ownership and this PR would make it more ambiguous.